Zend/zend_fibers.c

```diff
@@ -445,6 +445,23 @@ static void zend_fiber_object_free(zend_object *object)
 	zend_object_std_dtor(&fiber->std);
 }
 
+ZEND_API zend_fiber *zend_fiber_create(const zend_fcall_info *fci, const zend_fcall_info_cache *fci_cache)
+{
+	zend_fiber *fiber = (zend_fiber *) zend_fiber_object_create(zend_ce_fiber);
+	zval object;
+
+	fiber->fci = *fci;
+	fiber->fci_cache = *fci_cache;
+
+	Z_TRY_ADDREF(fiber->fci.function_name);
+
+	ZVAL_OBJ(&object, &fiber->std);
+	GC_ADDREF(&fiber->std);
+	zval_ptr_dtor(&object);
```

Comment:
Uh, what's the point of this? You're adding a ref and then dropping it in the next line.

Reply:
I think I'm not fully grasping how the GC tracks objects that need to be deallocated.

```c
zend_fiber *fiber = zend_fiber_create(fci, fci_cache);
GC_DELREF(&fiber->std); // Leaks if zend_object not assigned to zval prior
```

If I remove the zval assignment, the object always leaks. What's the proper way to do this?